org.elastic.jdt.ls.core/src/org/elastic/jdt/ls/core/internal/SecurityPermissionHelper.java

```diff
@@ -0,0 +1,115 @@
+	
+	// Deny all bundles to execute files
+	private static final String DENY_EXEC_PERM = 
+ 	       "DENY {" +
+ 		   "( java.io.FilePermission \"<<ALL FILES>>\" \"execute\" )" +
+ 		   "} \"Deny permission\"";
+	
+	// permissions for Maven and Gradle bundles
+	private static final String MINIMAL_PERM_TEMPLATE = 
+ 	        "ALLOW {" +
+			"( org.osgi.framework.AdminPermission \"*\" \"*\" )" + 	
+			"( org.osgi.framework.AdaptPermission \"*\" \"adapt\" )" +
+			"( org.osgi.framework.PackagePermission \"*\" \"exportonly,import\" )" + 
+			"( org.osgi.framework.ServicePermission \"*\" \"get,register\")" + 
+			"( org.osgi.framework.BundlePermission \"*\" \"host,provide,fragment\")" +
+			"( org.osgi.service.application.ApplicationAdminPermission \"*\" \"lifecycle, lock, schedule\" )" +
+			"( org.eclipse.equinox.log.LogPermission \"*\" \"log\" )" +
+			"( java.net.SocketPermission \"%s\" \"connect,resolve\" )" +
+			"( java.net.SocketPermission \"localhost\" \"resolve\" )" +
+			"( java.net.NetPermission \"getNetworkInformation\" )" +  
+			"( java.lang.reflect.ReflectPermission \"suppressAccessChecks\" )" +
+			"( java.io.FilePermission \"<<ALL FILES>>\" \"read\" )" + 	
+			"( java.io.FilePermission \"%s\" \"write,delete\" )" + 
+			"( java.io.SerializablePermission \"enableSubstitution\" )" +
+			"( java.lang.RuntimePermission \"*\" \"*\" )" + 
+			"( java.util.PropertyPermission \"*\" \"read, write\" )" +
+			"( java.net.NetPermission \"getProxySelector\" )" +
+			"%s } \"Normal bundles\"";
+	
+	private SecurityPermissionHelper() {}
+	
+	public static void setSecurityPermissions(BundleContext bundleContext) {
+		
+                ServiceReference sRef = bundleContext.getServiceReference(ConditionalPermissionAdmin.class.getName());
+                
+                // Get hold of OSGi ConditionalPermissionAdmin service...
+                ConditionalPermissionAdmin conPermAdmin = (ConditionalPermissionAdmin) bundleContext.getService(sRef);
+		
+                // Create new "atomic rules update" object
+                ConditionalPermissionUpdate update = conPermAdmin.newConditionalPermissionUpdate();
+
+                // Get list of existing permissions (normally null)
+                List<ConditionalPermissionInfo> perms = update.getConditionalPermissionInfos();
+                
+                // Clear old permissions
+                update.getConditionalPermissionInfos().clear();
+                
+                ConditionalPermissionInfo coreP = conPermAdmin.newConditionalPermissionInfo(CORE_PERM);
+                perms.add(coreP);
+                ConditionalPermissionInfo denyP = conPermAdmin.newConditionalPermissionInfo(DENY_EXEC_PERM);
+                perms.add(denyP);
+                String minimalPerm = String.format(
+                                MINIMAL_PERM_TEMPLATE,
+                                Environment.get(CLIENT_HOST) + ":" + Environment.get(CLIENT_PORT),
+                                getDataFolder(),
```

Comment:
Should we escape these parameters?
For example, could the folder path contain at special characters?

Reply:
We don't use these parameters in executable command.

Reply:
what if there is a quotation mark '"' in the  name of the data folder, could it make the rest of the file to be invalid?

Reply:
Actually the folder here is the data folder of kibana, which is set when launching kibana, not from any end users. So I think special characters like quotes are unlikely to appear in the data folder.

Reply:
I understand, in fact the attacker could also be the system admin, who may leave a vulnerability for further use. Anyway, it could be a good habit to escape parameters properly when generating strings dynamically.

Reply:
Done.